### PR TITLE
feat: Implement tests for Units of Measure(UoM)

### LINKED ISSUE #684: Implement the automation tests for UoM feature
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]

 This feature request is for core-metadata

### Description [**REQUIRED**]
This is a placeholder for implementing test for
https://docs.edgexfoundry.org/2.2/design/adr/core/0022-UoM/

There will be more details once this ADR is implemented

### PR DESCRIPTION
**Pull request by cherrycl**
1. Implement tests for Units of Measure(UoM)
2. Remove tests on PATCH device resource, device resource can't update units.
3. Modify some keywords

Fix #684

Signed-off-by: Cherry Wang <cherry@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Comment by cherrycl** [comment]
Raised issue https://github.com/edgexfoundry/edgex-go/pull/4140 for test `ProfileResourcePOST003`

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive.robot [hunk 1]
```diff
@@ -54,26 +53,24 @@ ProfilePOST004 - Create device profile by upload file without deviceResources an
...
+    Given Update Service Configuration On Consul  ${uomValidationPath}  true
+    When Modify Device Profile Test-Profile-1 With valid Units Value
     Then Should Return Status Code "201"
     And Should Return Content-Type "application/json"
     And Should Contain "id"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    [Teardown]  Run Keywords  Set UoM Validation to False
-    ...                  AND  Delete Device Profile
+    [Teardown]  Run Keywords  Update Service Configuration On Consul  ${uomValidationPath}  false
+    ...                  AND  Delete Device Profile By Name  Test-Profile-1
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive.robot:73]: Should also delete NEW-Test-Profile-1.yaml

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PUT-Positive.robot [hunk 2]
```diff
@@ -53,29 +53,31 @@ ProfilePUT003 - Update device profiles by upload file
...
     Then Should Return Status Code "200"
     And Should Return Content-Type "application/json"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And Profile Should Be Updated
-    [Teardown]  Run Keywords  Set UoM Validation to False
-    ...                  AND  Delete Device Profile
+    And Resource Units Should Be Updated in Test-Profile-2
+    [Teardown]  Run Keywords  Update Service Configuration On Consul  ${uomValidationPath}  false
+    ...                  AND  Delete Device Profile By Name  Test-Profile-2
+    ...                  AND  Delete Profile Files  NEW-Test-Profile-1.yaml
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PUT-Positive.robot:80]:
```suggestion
    ...                  AND  Delete Profile Files  NEW-Test-Profile-2.yaml
```

File: TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot [hunk 3]
```diff
@@ -755,4 +753,63 @@ Create A Provision Watcher Sample With Associated Test-Device-Service And ${devi
...
+Set Profile Units Value To ${valid}
+    Run Keyword If  "${valid}" == "valid"  Retrieve Valid Units Value
+    FOR  ${INDEX}  IN RANGE  len(${deviceProfile}[0][profile][deviceResources])
+        ${unit}  Run Keyword If  "${valid}" == "valid"  Evaluate  random.choice(@{uom_units})  random
+                 ...       ELSE  Set Variable  invalid
+        Set To Dictionary   ${deviceProfile}[0][profile][deviceResources][${INDEX}][properties]  units=${unit}
+    END
+    Set Test Variable  ${deviceProfile}  ${deviceProfile}
+
+Create A Device Profile With ${valid} Units Value ${file}
```
[the remark this answers is not in the document]
Comment (cherrycl) [reply TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot]: Using `Create A Device Profile ${file} With ${valid} Units Value` is conflict with `Create A Device Profile Sample`, change to `Create A Profile ${file} With ${valid} Units Value`

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive.robot (hunk 1 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive.robot:73]: Updated

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PUT-Positive.robot (hunk 2 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PUT-Positive.robot]: Updated

File: TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot [hunk 4]
```diff
@@ -510,28 +511,25 @@ Delete Profile Files
...
-    ${json_string}=  Evaluate  json.loads(r'''${resource_data}''')  json
-    Generate resource  ${json_string}
+    Generate deviceResource
     Set To Dictionary  ${resourceProfile}[1]  profileName=Test-Profile-2
     Set To Dictionary  ${resourceProfile}[2]  profileName=Test-Profile-3
 
-Generate resource
-    [Arguments]  ${data_list}
-    ${len}  Get Length  ${data_list}
+Generate deviceResource
```
[the remark this answers is not in the document]
Comment (cherrycl) [reply TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot]: Updated

**Review by jumpingliu — APPROVED** [review]
LGTM

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**